tests/db_functions/test_nullif.py

```diff
@@ -0,0 +1,33 @@
+from unittest import skipIf, skipUnless
+
+from django.db import connection
+from django.db.models import Value
+from django.db.models.functions import NullIf
+from django.test import TestCase
+
+from .models import Author
+
+
+class NullIfTests(TestCase):
+    def setUpTestData(cls):
+        Author.objects.create(name='John Smith', alias='smithj')
+        Author.objects.create(name='Rhonda', alias='Rhonda')
+
+    @skipIf(connection.vendor == 'Oracle', "Oracle doesn't use NULL, but the empty string")
```

Comment:
What is it that causes Oracle to return an empty string?

According to their [documentation](https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions102.htm) this shouldn't be the case:

> If they are equal, then the function returns null.

It sounds like there is either a problem in Django or cx_Oracle?

If this is going to remain like this, either we need a workaround for Oracle to force `NULL` or it should be documented.

Reply:
@pope1ni It's a convention rather than an issue, please see [#null-and-empty-strings](https://docs.djangoproject.com/en/dev/ref/databases/#null-and-empty-strings) documentation and this [code](https://github.com/django/django/blob/master/django/db/backends/oracle/operations.py#L159-L167). If we will use fields with types other than `BinaryField` and `CharField` then `NullIf` should return `None`, e.g:
```python
from decimal import Decimal

from django.db.models.functions import NullIf
from django.test import TestCase

from .models import DecimalModel

class NullIfTests(TestCase):

    @classmethod
    def setUpTestData(cls):
        DecimalModel.objects.create(n1=Decimal('1.1'), n2=Decimal('1.3'))
        DecimalModel.objects.create(n1=Decimal('1.2'), n2=Decimal('1.2'))

    def test_nullif(self):
        decimals = DecimalModel.objects.annotate(nullif=NullIf('n1', 'n2')).values_list('nullif')
        self.assertSequenceEqual(decimals, [(Decimal('1.1'),), (None,)])
```

Reply:
I see. Wasn't aware of that quirk as I don't use Oracle - thanks for the explanation.

I think in that case, a test with another type such as the example you provided should definitely be added, and the documentation for `NullIf` should highlight this still substitution applies to it's output - after all, one typically expects the value from this function to be `None` or the first argument.

Reply:
I'll put a link to that section in the documentation in this patch.